neutron/neutron.go

```diff
@@ -160,12 +161,15 @@ type FloatingIPV2 struct {
 }
 
 // ListFloatingIPsV2 lists floating IP addresses associated with the tenant or account.
-func (c *Client) ListFloatingIPsV2() ([]FloatingIPV2, error) {
+func (c *Client) ListFloatingIPsV2(filter ...*Filter) ([]FloatingIPV2, error) {
 	var resp struct {
 		FloatingIPV2s []FloatingIPV2 `json:"floatingips"`
 	}
-
-	requestData := goosehttp.RequestData{RespValue: &resp}
+	var params *url.Values
+	if len(filter) > 0 {
+		params = &filter[0].v
```

Comment:
If we're only ever going to use just one Filter value we should return an error if len(filter) > 1 (and add a test)
Otherwise it's confusing reading the code.
The doc comment also should be updated to mention the filter and expected semantics.

The other option:

ListFloatingIPsV2(filter *Filter) ([]FloatingIPV2, error)

if filter != nil {
  params = &filter.v
}

But this will change the caller to make it pass in nil when there's no filter. We have done that in other places, so whatever works best....

Reply:
The filter for ListFloatingIPsV2() is working the same as for ListNetworksV2().  I did update the comments.